omod/src/main/webapp/resources/htmlforms/hts/htsInitialTest.html

```diff
@@ -677,7 +681,154 @@
                 getField('hiv-final-result.value').val(1138);
             }
 
-        });
+        // add code for ML feature
+
+    // handle ML prediction button
+	jq("#btn-ml-prediction").click(function () {
+        var ctxPath = typeof openmrsContextPath === 'string' ? openmrsContextPath : OPENMRS_CONTEXT_PATH;
+		var latestMaritalStatus = "<lookup expression="fn.latestObs(1054).getValueCoded()"/>";
+		var monthsSinceLastTest = getValue('duration_months.value');
+		var populationType = getValue('population-type.value');
+		var isDisabled = getValue('has-disability.value');
+		var everTested = getValue("ever-tested.value");
+		var testedAs = getValue("couple-testing.value");
+		var entryPoint = getValue("hts-entry-point.value");
+		var testStrategy = getValue("strategy.value");
+		var tbScreening = getValue("tb-screening.value");
+		var selfTested = getValue("hb-self-test.value");
+
+		var predictionVariables = {
+			AgeAtTest: pAge,
+			MonthsSinceLastTest: monthsSinceLastTest,
+			Gender_Male: pgender == 'M' ? 1 : 0,
+			KeyPopulationType_GP: 0,
+			KeyPopulationType_SW: 0,
+			MaritalStatus_Married: 0,
+			MaritalStatus_Divorced: 0,
+			MaritalStatus_Polygamous: 0,
+			MaritalStatus_Widowed: 0,
+			MaritalStatus_Minor: 0,
+			PatientDisabled_Not_Disabled: 0,
+			EverTestedForHIV_Yes: 0,
+			ClientTestedAs_Individual: 0,
+			EntryPoint_VCT: 0,
+			EntryPoint_OPD: 0,
+			EntryPoint_MTC: 0,
+			EntryPoint_IPD: 0,
+			EntryPoint_MOBILE: 0,
+			EntryPoint_Other: 0,
+			EntryPoint_HB: 0,
+			EntryPoint_PEDS: 0,
+			TestingStrategy_VCT: 0,
+			TestingStrategy_HB: 0,
+			TestingStrategy_MOBILE: 0,
+			TestingStrategy_HP: 0,
+			TestingStrategy_NP: 0,
+			TBScreening_No_Presumed_TB: 0,
+			ClientSelfTested_No: 0
+		};
+
+        // convert marital status
+		if (latestMaritalStatus == 5555) { // married monogamous
+			predictionVariables.MaritalStatus_Married = 1;
+		} else if (latestMaritalStatus == 159715) { // married polygamous
+			predictionVariables.MaritalStatus_Polygamous = 1;
+		} else if (latestMaritalStatus == 1058) { // divorced
+			predictionVariables.MaritalStatus_Divorced = 1;
+		} else if (latestMaritalStatus == 1058) { // divorced
```

Comment:
Is maritalstatus divorced(1058) not repeated?

Reply:
I can't find the repeated line in that file. Try viewing the changes and point me to it.

Reply:
@ojwanganto It was on the first commit, but I see you came and fixed it in this commit "integration of ML casefinding model in HTS forms"